processor/src/main/java/org/mapstruct/ap/internal/model/beanmapping/SourceReference.java

```diff
@@ -384,4 +402,23 @@ public SourceReference pop() {
         }
     }
 
+    public List<SourceReference> push(TypeFactory typeFactory, Method method ) {
```

Comment:
Is `push` the right method name? I thought it changes the `SourceReference`, but it creates a new one it seems

Reply:
it does the opposite of pop.. So it adds an element. As a consequence it generates multiple new source references.